language-server/language-server-api/.js/src/main/scala/aqua/lsp/TypeJs.scala

```diff
@@ -0,0 +1,105 @@
+package aqua.lsp
+
+import aqua.types.*
+
+import scala.scalajs.js.Dictionary
+import scala.scalajs.js.JSConverters.*
+import scalajs.js
+
+sealed trait TypeJs extends js.Object {
+  val tag: String
+}
+
+class ScalarTypeJs(val name: String) extends TypeJs {
+  val tag: String = "scalar"
+}
+
+class ArrayTypeJs(val element: TypeJs) extends TypeJs {
+  val tag: String = "array"
+}
+
+class OptionTypeJs(val element: TypeJs) extends TypeJs {
+  val tag: String = "option"
+}
+
+class StreamTypeJs(val element: TypeJs) extends TypeJs {
+  val tag: String = "stream"
+}
+
+class StreamMapTypeJs(val element: TypeJs) extends TypeJs {
+  val tag: String = "streammap"
+}
+
+class CanonStreamTypeJs(val element: TypeJs) extends TypeJs {
+  val tag: String = "canon"
+}
+
+class AbilityTypeJs(val name: String, val fields: js.Dictionary[TypeJs]) extends TypeJs {
+  val tag: String = "ability"
+}
+
+class StructTypeJs(val name: String, val fields: js.Dictionary[TypeJs]) extends TypeJs {
+  val tag: String = "struct"
+}
+
+class ServiceTypeJs(val name: String, val fields: js.Dictionary[TypeJs]) extends TypeJs {
+  val tag: String = "service"
+}
+
+trait ProductType extends TypeJs
+
+class LabeledConsTypeJs(val args: js.Dictionary[TypeJs]) extends TypeJs {
+  val tag: String = "labeled"
+}
+
+class UnlabeledConsTypeJs(val types: js.Array[TypeJs]) extends TypeJs {
+  val tag: String = "unlabeled"
+}
+
+class ArrowTypeJs(val domain: LabeledConsTypeJs, val codomain: UnlabeledConsTypeJs) extends TypeJs {
+  val tag: String = "arrow"
+}
+
+class NilTypeJs extends TypeJs {
+  val tag: String = "nil"
+}
+
+class BottomTypeJs extends TypeJs {
+  val tag: String = "bottom"
+}
+
+class TopTypeJs extends TypeJs {
+  val tag: String = "top"
+}
+
+object TypeJs {
+
+  def typeList(types: List[(String, Type)]): Dictionary[TypeJs] =
```

Comment:
nit: Could accept `Iterable` and eliminate the need to convert to `List` on call site